### PR TITLE
Breaking Change: Add uri parameter to getRequestMetadata.

### PR DESCRIPTION
**Pull request by anthmgoogle**
Applying a breaking change from the specification that is required to enable the JwtAccess credential implementation.

### CODE REVIEW

**Comment by ejona86** [comment]
This is going to be a bit of a problem to implement on the gRPC side, as we don't know the scheme and we normally won't know the host with the current implementation. @louiscryan, do you have ideas for handling the new API in gRPC?

**Comment by anthmgoogle** [comment]
For reference @tbetbetbe and Julien Boeuf have implemented equivalent plumbing for Ruby and C++. We need at least the uri to construct the JwtAccess/IdToken header.

File: credentials/java/com/google/auth/Credentials.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
    *
    * @throws IOException if there was an error getting up-to-date access.
    */
-  public abstract Map<String, List<String>> getRequestMetadata() throws IOException;
+  public abstract Map<String, List<String>> getRequestMetadata(String uri) throws IOException;
```
Comment (ejona86) [inline credentials/java/com/google/auth/Credentials.java]: Instead of passing in a String, should this be a URI?

**Comment by anthmgoogle** [comment]
Updated to use URI instead of String.

**Comment by ejona86** [comment]
Do you think URI makes more sense? I am okay either way, but wanted to pose the question.

@anthmgoogle, LGTM

**Comment by anthmgoogle** [comment]
Yes, I think URI works better for Java.
I'll wait until we confirm that we will be able to pass this in before merging.

**Pull request closed without merging**